src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -143,10 +143,8 @@ private static async Task<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(St
                 cancellationToken.ThrowIfCancellationRequested();
 
                 // active file diagnostics:
-                if (documentId != null && kind != AnalysisKind.NonLocal && stateSet.TryGetActiveFileState(documentId, out var state))
-                {
-                    return state.GetAnalysisData(kind).Items;
-                }
+                if (documentId != null && kind != AnalysisKind.NonLocal && stateSet.IsActiveFile(documentId))
+                    return [];
```

Comment:
the prior logic would get the active file state if it exsits, and get the items off of it.  but we know that that items was always empty (since nothing ever called Save), so we can shortcircuit this to just returning empty if 'IsActiveFile' returns true.

note: i have no idea waht this code is for, and it likely is problematic in some way.